cloud/azure/bin/lib/key_vault.sh

```diff
@@ -124,5 +124,5 @@ function key_vault::has_secret() {
     --query value \
     -o tsv 2>&1 >/dev/null)"
 
-  echo "${SECRET_RESULT}" | grep -q -v "SecretNotFound"
+  echo ${SECRET_RESULT}
```

Comment:
The old expression doesn't work unfortunately. It always returns true regardless of whether or not "SecretNotFound" is in the string. I don't know enough bash to understand why it's broken. I decided to fix it by writing an expression that I knew would always work because it's a direct check against the returned string.

Reply:
Hmm the old expression works in my shell - I'd be happy to help you debug it! I have a lot of experience with bash :| 

> $ echo "bloo blah" | grep -q -v "SecretNotFound"; echo $?
> 0
> $ echo "bloo blah SecretNotFound" | grep -q -v "SecretNotFound"; echo $?
> 1

If you prefer the new logic can you move it into this function so that it still returns true/false and isn't repeated at the callsite? Either way happy to look at it with you!

Reply:
Ah! I didn't realize you had to add echo $? at the end of the expression to get it to work. That's the missing piece, will add back this expression and revert my changes in the other files

Reply:
The reason I changed it is because the current expression doesn't work. The if statement doesn't evaluate as expected, it always returns "true" and never creates the secret even if there is no secret key in the store. I found this out after the script kept on failing on this step and some debug statements pinpointed the error to this step in the process.

No secrets in the keyvault. Script will create since no secrets are present
<img width="954" alt="Screenshot 2024-08-20 at 4 12 32 PM" src="https://github.com/user-attachments/assets/3e137a20-f5a1-4478-87d3-1f3a2b78ab18">

Script erroneously believes that secrets exist 
<img width="502" alt="Screenshot 2024-08-20 at 4 12 44 PM" src="https://github.com/user-attachments/assets/cfda38f6-2605-49c2-b094-5baad09c7544">

This is because the condition `if key_vault::has_secret "${vault}" "${id}"` always evaluates to true. 

When I evaluate the return expression `echo "${SECRET_RESULT}" | grep -q -v "SecretNotFound"` in my terminal I actually get empty string return. 
<img width="873" alt="Screenshot 2024-08-20 at 4 24 09 PM" src="https://github.com/user-attachments/assets/009098a0-bed8-4f52-8dd3-f126c6f2a1c9">

Reply:
Surprisingly adding echo $ so that the return expression looks like this ` echo "${SECRET_RESULT}" | grep "SecretNotFound" | echo $` does fix things even though it comes with extra new lines  

<img width="398" alt="Screenshot 2024-08-20 at 4 38 06 PM" src="https://github.com/user-attachments/assets/54be8dd6-5b21-4004-9f02-b98807cf98bd">

Reply:
This doesn't work - it just always returns not found. I scheduled some time tomorrow to pair on it :)

Reply:
I think echo $ makes it so that it looks like the vault never has the secret, something about the current expression `echo "${SECRET_RESULT}" | grep -q -v "SecretNotFound"` needs to be fixed. 

I'm going to try your suggestion nick and see if it works.

Reply:
Summarizing our offline debug session for posterity: the SECRET_RESULT includes multiple lines, and if the secret is not found only some of those lines include SecretNotFound. So grep -v SecretNotFound always returns true regardless of whether the secret is not found.

Nick's solution works, because it inverts the result of the regular grep instead of using grep -v, but I think this would be more readable:

! echo "${SECRET_RESULT}" | grep -q "SecretNotFound"